shardingsphere-kernel/shardingsphere-data-pipeline/shardingsphere-data-pipeline-api/src/main/java/org/apache/shardingsphere/data/pipeline/api/task/progress/IncrementalTaskProgress.java

```diff
@@ -17,24 +17,49 @@
 
 package org.apache.shardingsphere.data.pipeline.api.task.progress;
 
-import lombok.AllArgsConstructor;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 import lombok.Getter;
-import lombok.NoArgsConstructor;
+import lombok.RequiredArgsConstructor;
 import lombok.Setter;
 import lombok.ToString;
 import org.apache.shardingsphere.data.pipeline.api.ingest.position.IngestPosition;
 
 /**
  * Incremental task progress.
  */
-@NoArgsConstructor
-@AllArgsConstructor
+@RequiredArgsConstructor
 @Getter
 @Setter
 @ToString
 public final class IncrementalTaskProgress implements TaskProgress {
     
-    private volatile IngestPosition<?> position;
+    private final Map<String, IncrementalTaskProgressItem> incrementalTaskProgressItemMap;
     
-    private IncrementalTaskDelay incrementalTaskDelay = new IncrementalTaskDelay();
```

Comment:
Could we keep the similar structure as YamlIncrementalTaskProgress?

Reply:
I find the  definition of `RuleAlteredJobContext`  as follows，so i changed it
```java
private final Collection<InventoryTask> inventoryTasks = new LinkedList<>();
    
private final Collection<IncrementalTask> incrementalTasks = new LinkedList<>();
```